protostar/cheatable_starknet/callable_hint_locals/mock_call_hint_local.py

```diff
@@ -0,0 +1,26 @@
+from typing import Callable
+
+from protostar.cheatable_starknet.controllers.contracts import ContractsController
+from protostar.starknet import Address, Selector
+
+from .callable_hint_local import CallableHintLocal
+
+
+class MockCallHintLocal(CallableHintLocal):
+    def __init__(self, controller: ContractsController) -> None:
+        super().__init__()
+        self._controller = controller
+
+    @property
+    def name(self) -> str:
+        return "mock_call"
+
+    def _build(self) -> Callable:
+        def mock_call(target_address: int, entrypoint: str, response: list[int]) -> int:
+            return self._controller.mock_call(
+                target_address=Address.from_user_input(target_address),
+                entrypoint=Selector(entrypoint),
+                response=response,
+            )
```

Comment:
Why not use previous pattern with returning `self.mock_call` method?

Reply:
I noticed that if you provide wrong number of arguments, Python will tell something similar to: "provided 2 arguments but expected 4".
The problem is it counts `self` as argument, resulting in a misleading message to the user.

Reply:
It's not going to be a problem in Cairo 1

Reply:
It may be in case of bugs on the Rust side. Something feels off with returning methods instead of functions.

Reply:
yes, the code will crash if we implement it wrong, however this will not have an effect for the user since i think we're going to test it before releasing.

Reply:
When returning a method, a user doesn't have to provide the `self` argument for some reason. I prefer to return a plain function to eliminate this magic behavior, especially returning methods doesn't provide any benefits.

Reply:
I actually think we should introduce stop_mock_call instead, we will be adapting it to cairo 1 in the next few days

Reply:
Agreed - this seems to be out of scope of this PR though

Reply:
Please leave a task for adjusting the implementations accordingly though, for consistency

Reply:
It's new code. I think it would be out of scope if I would modify existing hint local.

Reply:
I think we should adjust it here, we shouldn't return higher order functions anywhere.

Reply:
[return method](https://github.com/software-mansion/protostar/pull/1460/commits/82d839507f8968690eaee469aeaeda7c9c5d5af7)

Reply:
https://github.com/software-mansion/protostar/issues/1472